pkg/webhook/policy.go

```diff
@@ -501,14 +508,83 @@ func (h *validationHandler) reviewRequest(ctx context.Context, req *admission.Re
 		}
 	}
 
+	review, err := h.createReviewForRequest(ctx, req)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create augmentedReview: %s", err)
+	}
+
+	// Convert the request's generator resource to unstructured for expansion
+	obj := &unstructured.Unstructured{}
+	if _, _, err := deserializer.Decode(req.Object.Raw, nil, obj); err != nil {
+		return nil, fmt.Errorf("error decoding generator resource %s: %s", req.Name, err)
+	}
+	obj.SetNamespace(req.Namespace)
+	obj.SetGroupVersionKind(
+		schema.GroupVersionKind{
+			Group:   req.Kind.Group,
+			Version: req.Kind.Version,
+			Kind:    req.Kind.Kind,
+		})
+
+	// Expand the generator and apply mutators to the resultant resources
+	base := &mutationtypes.Mutable{
+		Object:    obj,
+		Namespace: review.Namespace,
+		Username:  req.AdmissionRequest.UserInfo.Username,
+		Source:    mutationtypes.SourceTypeGenerated,
+	}
+	resultants, err := h.expansionSystem.Expand(base)
```

Comment:
should we check ShouldExpand before calling Expand here?

Reply:
Expand resultants will be an empty slice if no ExpansionTemplates match with no additional work compared to calling ShouldExpand first, so it's not necessary. I felt this simplified the code a bit by reducing control flow.